exchange/exchange_test.go

```diff
@@ -1830,6 +1830,18 @@ func mockHandler(statusCode int, getBody string, postBody string) http.Handler {
 	})
 }
 
+func mockSlowHandler(statusCode int, getBody string, postBody string) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		time.Sleep(500 * time.Millisecond)
+		w.WriteHeader(statusCode)
+		if r.Method == "GET" {
```

Comment:
I don't think the method type is important for the test. Consider accepting a single `body string` and always return it.